### PR TITLE
Add cleanup and stats commands to run_doctype_migration

### PR DESCRIPTION
**Pull request by dannyroberts**
depends on
- https://github.com/dimagi/commcare-hq/pull/8389
- https://github.com/dimagi/commcare-hq/pull/8387 (reindex)
- https://github.com/dimagi/commcare-hq/pull/8386
- https://github.com/dimagi/commcare-hq/pull/8385

(no separate reindex of it's own after 8387 is done)

`--cleanup` deletes docs from the source db (it'll prompt you to make sure you've copied them to the target db)
`--stats` prints out something like this:

```
           doc_type             Source  Target
AdminUserRole                   0   0
AdminUserRole-Deleted           0   0
CommCareUser                    0   142
CommCareUser-Deleted            0   0
DeleteGroupRecord               0   15
DeleteGroupRecord-Deleted       0   0
DomainInvitation                0   0
DomainInvitation-Deleted        0   0
DomainRemovalRecord             0   0
DomainRemovalRecord-Deleted     0   0
Group                           0   35
Group-Deleted                   0   15
Invitation                      0   11
Invitation-Deleted              0   0
OrgRemovalRecord                0   0
OrgRemovalRecord-Deleted        0   0
UserRole                        0   72
UserRole-Deleted                0   0
WebUser                         0   17
WebUser-Deleted                 0   0
```

### CODE REVIEW

**Comment by dannyroberts** [comment]
@snopoke cc: @czue

File: corehq/doctypemigrations/management/commands/run_doctype_migration.py [hunk 1]
```diff
@@ -79,3 +113,17 @@ def handle_continuous(self, migrator):
...
+
+    def handle_cleanup(self, migrator):
+        migrator.phase_3_clean_up()
+
+    def handle_stats(self, migrator):
+        [(source_db, source_counts),
+         (target_db, target_counts)] = migrator.get_doc_counts()
+        self.stdout.write('Source DB: {}\n'.format(source_db.uri))
+        self.stdout.write('Target DB: {}\n'.format(target_db.uri))
+        self.stdout.write('{:^30}\tSource\tTarget\n'.format('doc_type'))
```
Comment (snopoke) [inline corehq/doctypemigrations/management/commands/run_doctype_migration.py]: Why self.stdout.write and not just print?

Comment (dannyroberts) [inline corehq/doctypemigrations/management/commands/run_doctype_migration.py]: It seems to be the convention is all https://docs.djangoproject.com/en/1.7/howto/custom-management-commands/#management-commands-output. Although apparently the newlines aren't necessary as I just learned from reading that.

File: corehq/apps/domain/_design/views/related_to_domain/map.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+function(doc) {
```
Comment (dimagimon) [inline corehq/apps/domain/_design/views/related_to_domain/map.js]: Missing name in function declaration.

**Comment by czue** [comment]
will let @snopoke review/merge.

**Pull request merged**